fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java

```diff
@@ -62,17 +63,10 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
             }
         }));
 
-        // Join Operator could not accept an output column, so try to choose a column for it
-        LogicalOperator inputOp = (LogicalOperator) input.inputAt(0).getOp();
-        if (newMap.isEmpty() && inputOp.getOpType().equals(OperatorType.LOGICAL_JOIN)) {
-            ColumnRefSet candidates = new ColumnRefSet(projectOperator.getColumnRefMap().keySet());
-            ColumnRefOperator smallestColumn =
-                    inputOp.getSmallestColumn(candidates, context.getColumnRefFactory(), input.inputAt(0));
-            if (smallestColumn != null) {
-                ScalarOperator expr = projectOperator.getColumnRefMap().get(smallestColumn);
-                newMap.put(smallestColumn, expr);
-                requiredInputColumns.union(smallestColumn);
-            }
```

Comment:
Why not handle `PruneProjectRule`?  `getSmallestColumn` can be removed ?

Reply:
If we have add the tinyint col when projectMap is empty in `PruneProjectColumnsRule`, I think the logic of filling smallest col in `PruneProjectRule` can be removed. I want to test this code for sometimes, if it's bugfree then remove the `PruneProjectRule`.